File: binding/src/spatialaudio/libqaudiolistener.cpp
#include <QAudioEngine>
#include <QAudioListener>
#include <QObject>
#include <QQuaternion>
#include <QVector3D>
#include <qaudiolistener.h>
#include "libqaudiolistener.h"
#include "libqaudiolistener.hxx"

QAudioListener* QAudioListener_new(QAudioEngine* engine) {
	 return new QAudioListener(engine);
}

void QAudioListener_SetPosition(QAudioListener* self, QVector3D* pos) {
	self->setPosition(*pos);
}

QVector3D* QAudioListener_Position(const QAudioListener* self) {
	return new QVector3D(self->position());
}

void QAudioListener_SetRotation(QAudioListener* self, const QQuaternion* q) {
	self->setRotation(*q);
}

QQuaternion* QAudioListener_Rotation(const QAudioListener* self) {
	return new QQuaternion(self->rotation());
}

QAudioEngine* QAudioListener_Engine(const QAudioListener* self) {
	return self->engine();
}

void QAudioListener_Delete(QAudioListener* self) {
    delete self;
}

File: binding/src/spatialaudio/libqaudiolistener.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qaudiolistener.html
pub const qaudiolistener = struct {

    /// New constructs a new QAudioListener object.
    pub fn New(engine: ?*anyopaque) QtC.QAudioListener {
        return qtc.QAudioListener_new(@ptrCast(engine));
    }


    pub fn SetPosition(self: ?*anyopaque, pos: QtC.QVector3D) void {
        qtc.QAudioListener_SetPosition(@ptrCast(self), @ptrCast(pos));
    }

    pub fn Position(self: ?*anyopaque, ) QtC.QVector3D {
        return qtc.QAudioListener_Position(@ptrCast(self));
    }

    pub fn SetRotation(self: ?*anyopaque, q: ?*anyopaque) void {
        qtc.QAudioListener_SetRotation(@ptrCast(self), @ptrCast(q));
    }

    pub fn Rotation(self: ?*anyopaque, ) QtC.QQuaternion {
        return qtc.QAudioListener_Rotation(@ptrCast(self));
    }

    pub fn Engine(self: ?*anyopaque, ) QtC.QAudioEngine {
        return qtc.QAudioListener_Engine(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAudioListener_Delete(@ptrCast(self));
    }
};
